### PR TITLE
Logfile

### PR DESCRIPTION
**Pull request by MirijaHa**
This PR writes detailed log messages on lost companies to a logfile in the output dir, only basic information that reveal that the code is running are printed to console.
In a follow up I will at timetamps to log files and results, so they do not overwrite each other.
FYI @jakubcervenka

### CODE REVIEW

**Review by maurolepore — APPROVED** [review]
LGTM. 

My suggestions:

1. The formatting "  -- {x}" and "  >> {x}" could be extracted in a lil helper, say `fmt_log1()` and `fmt_log2()`.
2. `paste_write()` with few changes could be happier

``` r
# x The name is a bit weird
# x The documentation is hard because I have to do it myself
#
# Blah blah
# @param ... Blah blah
# @param log_path Blah blah
paste_write <- function(..., log_path) {
  text <- paste(...)
  write(text, file = log_path, append = TRUE)
  invisible()
}

# v The name is a bit clearer
# v The documentation is easy because I can just pont to ?write
#
# A version of [write()] that is easy to experiment with and appends by default
# @inheritParams write
append_file <- function(x, file = "", append = TRUE, ...) {
  write(x, file = file, append = append, ...)
  invisible(x)
}

# x Experimenting is hard
try(paste_write("a"))
#> Error in cat(x, file = file, sep = c(rep.int(sep, ncolumns - 1), "\n"),  : 
#>   argument "log_path" is missing, with no default

# v Experimenting is easier
append_file("a")
#> a
append_file(paste("a", "b"))
#> a b
append_file(c("a", "b"))
#> a
#> b
c("a", "b") |> append_file()
#> a
#> b

# x The names suggests the second argument is the output directory but this fails
# because the dots force me to use name the argument for output explicitely
file <- tempfile()
try(
  "a" |> paste_write(file)
)
#> Error in write(text, file = log_path, append = TRUE) : 
#>   argument "log_path" is missing, with no default

# v Removing the dots, this now works as the name promises
file <- tempfile()
"a" |> append_file(file)
c("c", "d") |> append_file(file)
# Inspect
readLines(file)
#> [1] "a" "c" "d"
```

<sup>Created on 2021-11-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

**Comment by MirijaHa** [comment]
> LGTM.
> 
> My suggestions:
> 
> 1. The formatting "  -- {x}" and "  >> {x}" could be extracted in a lil helper, say `fmt_log1()` and `fmt_log2()`.
> 2. `paste_write()` with few changes could be happier:
> 
> ```r
> # x The name is a bit weird
> # x The documentation is hard because I have to do it myself
> #
> # Blah blah
> # @param ... Blah blah
> # @param log_path Blah blah
> paste_write <- function(..., log_path) {
>   text <- paste(...)
>   write(text, file = log_path, append = TRUE)
>   invisible()
> }
> 
> # v The name is a bit clearer
> # v The documentation is easy because I can just pont to ?write
> #
> # A version of [write()] that is easy to experiment with and appends by default
> # @inheritParams write
> append_file <- function(x, file = "", append = TRUE, ...) {
>   write(x, file = file, append = append, ...)
>   invisible(x)
> }
> 
> # x Experimenting is hard
> try(paste_write("a"))
> #> Error in cat(x, file = file, sep = c(rep.int(sep, ncolumns - 1), "\n"),  : 
> #>   argument "log_path" is missing, with no default
> 
> # v Experimenting is easier
> append_file("a")
> #> a
> append_file(paste("a", "b"))
> #> a b
> append_file(c("a", "b"))
> #> a
> #> b
> c("a", "b") |> append_file()
> #> a
> #> b
> 
> # x The names suggests the second argument is the output directory but this fails
> # because the dots force me to use name the argument for output explicitely
> file <- tempfile()
> try(
>   "a" |> paste_write(file)
> )
> #> Error in write(text, file = log_path, append = TRUE) : 
> #>   argument "log_path" is missing, with no default
> 
> # v Removing the dots, this now works as the name promises
> file <- tempfile()
> "a" |> append_file(file)
> c("c", "d") |> append_file(file)
> # Inspect
> write(readLines(file), "")
> #> a
> #> c
> #> d
> ```
> 
> Created on 2021-11-12 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)

implemented 1 and 2.1

File: R/calculate.R [hunk 1]
```diff
@@ -134,14 +137,14 @@ calculate_exposure_by_technology_and_company <- function(asset_type,
...
               exposure_by_technology_and_company_filtered$company_name)
       )
-    cat(
-      "      >> When filtering out holdings with exposures of 0 value, dropped rows for",
-      n_companies_pre - n_companies_post, "out of", n_companies_pre, "companies\n"
-    )
-    cat("        >> percent loss:", percent_loss, "\n")
-    cat("        >> affected companies:\n")
+    paste_write(
+      format_intend_1(), "When filtering out holdings with exposures of 0 value, dropped rows for",
```
Comment (jacobvjk) [inline R/calculate.R:141]: are we talking about the two different types of indents here? if so: spelling. If not: intent? still spelling.
Or something else?

**Review by jacobvjk — COMMENTED** [review]
see comment

File: R/calculate.R (hunk 1 above)
Comment (MirijaHa) [reply R/calculate.R:141]: I starred at intend for quite a while, knew sth was off but did not get what, fixing to indent

**Review by jacobvjk — APPROVED** [review]
lgtm

**Pull request merged**